### ISSUE TITLE
DatePicker no longer triggers Onvalidate for PageType=List

### ISSUE THREAD

**Issue by PieterJacob** [microsoft/ALAppExtensions#10615]
Since version 17.02 we encounter peculiar behavior when selecting dates via the DatePicker. After selecting a date via the date picker, the OnValidate is no longer triggered  when manually selecting another field by clicking on it, rather than using the Tab key. It looks like this behavior only happens when data is entered in a page of type list.

I have created a project like below to reproduce the problem.
1. Create a table with at least one Date field (or use existing table), and some other fields.
2. Create a page of type List for this table.

```
Table 50000 DatePickerTest
{    
    fields{
        field(10;PK;Integer)
        {
            AutoIncrement = true;
        }
        field(20; "Start Date"; Date)
        {
            trigger OnValidate()
            begin
                Message('Start Date OnValidate');
            end;
        }
        field(30; "End Date"; Date){}
        field(40; Description; Text[50]){}                
    }
}

Page 50006 DatePickerTestCard
{
    SourceTable = DatePickerTest;
    PageType = Card;
    layout{
        area(Content)
        {
            field("Start Date";"Start Date"){}
            field("End Date";"End Date"){}
            field(Description;Description){}
        }
    }
}
Page 50007 DatePickerList
{
    SourceTable = DatePickerTest;
    PageType = List;
    Editable = true;
    CardPageId = DatePickerTestCard;
    
    layout{
        area(Content)        
        {
            repeater(ListRepeater)
            {
                Field("Start Date";"Start Date"){}
                Field("End Date";"End Date"){}
                Field(Description;Description){}
            }
        }        
    }
}

```
To reproduce
1. Select a random date by using the Date Picker.
![image](https://user-images.githubusercontent.com/67948889/104916781-ce874180-5992-11eb-98fc-f7cd4e263ca0.png)
2. After the value is inserted into the field, jump to another field by clicking on it (tab key will correctly triggers OnValidate).
Enter some data.
![image](https://user-images.githubusercontent.com/67948889/104921591-d5fe1900-5999-11eb-90cf-9cca811c2175.png)
3. Note that no Message is shown defined in the OnValidate-trigger. 
4. Jump to a new row, and note that peculiar behavior occurs. 
![image](https://user-images.githubusercontent.com/67948889/104921640-e57d6200-5999-11eb-8fd8-6a53e417c643.png)

Expected behavior:
The OnValidate- trigger should always be triggered when a value is entered in the date field.

**Comment by StanPesotskiy** [comment]
This bug has already been fixed in December 2020 as part of 17.03 update. The hotfix was also ported down to 17.02.

**Comment by PieterJacob** [comment]
Great! Thanks. In that case we just have to wait for the next update.
